plugins/woocommerce/src/Internal/ProductFilters/FilterData.php

```diff
@@ -0,0 +1,331 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Automattic\WooCommerce\Internal\ProductFilters;
+
+use WC_Cache_Helper;
+use Automattic\WooCommerce\Internal\ProductFilters\Interfaces\QueryClausesGenerator;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class for filter counts.
+ */
+class FilterData {
+	/**
+	 * Instance of QueryClauses.
+	 *
+	 * @var QueryClausesGenerator
+	 */
+	private $query_clauses;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param QueryClausesGenerator $query_clauses Instance of QueryClausesGenerator.
+	 */
+	public function __construct( QueryClausesGenerator $query_clauses ) {
+		$this->query_clauses = $query_clauses;
+	}
+
+	/**
+	 * Get price data for current products.
+	 *
+	 * @param array $query_vars The WP_Query arguments.
+	 * @return object
+	 */
+	public function get_filtered_price( array $query_vars ) {
+		$pre_filter_counts = $this->pre_get_filter_counts( 'price', $query_vars, );
+
+		if ( isset( $pre_filter_counts ) ) {
+			return $pre_filter_counts;
+		}
+
+		$transient_key = $this->get_transient_key( $query_vars, 'price' );
+		$cached_data   = $this->get_cache( $transient_key );
+
+		if ( ! empty( $cached_data ) ) {
+			return $cached_data;
+		}
+
+		global $wpdb;
+
+		add_filter( 'posts_clauses', array( $this->query_clauses, 'add_query_clauses' ), 10, 2 );
+		add_filter( 'posts_pre_query', '__return_empty_array' );
+
+		$query_vars['no_found_rows']  = true;
+		$query_vars['posts_per_page'] = -1;
+		$query_vars['fields']         = 'ids';
+		$query                        = new \WP_Query();
+
+		$query->query( $query_vars );
+		$product_query_sql = $query->request;
```

Comment:
Seems unnecessary to assign this to a variable? You should be able to use `$query->request` directly later?

Reply:
Yes. This is copied over from Store API ProductQueryFilters. I agree that it's unnecessary. We will eventually (replace)[https://github.com/woocommerce/woocommerce/blob/c977084fc5395406fc1eb734266bf7b85baacc97/plugins/woocommerce/src/Internal/ProductFilters/FilterData.php#L89] sub query with cached product IDs. But I will still removed this unnecessary variable assignment.

Reply:
Addressed in 2423957094f3.